akka-docs-dev/rst/scala/code/docs/stream/cookbook/RecipeKeepAlive.scala

```diff
@@ -22,16 +22,16 @@ class RecipeKeepAlive extends RecipeSpec {
       val sink = Sink(sub)
 
       //#inject-keepalive
-      val keepAliveStream: Source[ByteString, Unit] = ticks
+      val keepAliveStream: Flow[Tick, ByteString, Unit] = Flow[Tick]
         .conflate(seed = (tick) => keepaliveMessage)((msg, newTick) => msg)
 
       val graph = FlowGraph.closed() { implicit builder =>
         import FlowGraph.Implicits._
         val unfairMerge = builder.add(MergePreferred[ByteString](1))
 
-        dataStream ~> unfairMerge.preferred
         // If data is available then no keepalive is injected
-        keepAliveStream ~> unfairMerge ~> sink
+        dataStream ~> unfairMerge.preferred
+        ticks ~> keepAliveStream ~> unfairMerge ~> sink
```

Comment:
I disagree, the keepAliveStream name exactly meant to mean that it is a stream of KeepAlive messages, i.e. it is a periodic source of ByteStrings that encode KeepAlive messages in the given format. As a Flow, this name looks wrong to me, it is now a stage that transforms ticks to KA messages, but no longer a stream of such messages.